### DISCUSSION TITLE
Using Durable Objects with Cloudflare Pages

### DISCUSSION THREAD

**Discussion by jhlabs** [honojs/hono#3325] [Q&A]
I followed an example to setup WebSockets with Hono + Durable Objects here https://github.com/honojs/hono/issues/3206#issuecomment-2269013293. I am using the Vite Cloudflare Pages setup for Hono. Now when I try to run this example I receive the following error: `Cannot find module 'cloudflare:workers'`. Are Durable Objects currently not supported when using Hono with Cloudflare Pages or am I just missing a dependency to make this work?

Thanks for your help!

**Comment by nakasyou** [comment]
Hi @jhlabs, what runtime such as wrangler, Bun or Deno do you use?

**Reply by jhlabs** [reply]
I have been following the steps outlined here https://hono.dev/docs/getting-started/cloudflare-pages with PNPM as my package manager. The way I understand this is that the Cloudflare relevant runtime is injected through the Vite plugins. I am not using wrangler in this case to start the development, but just vite.
